asst1/CODE/datastructs.c

```diff
@@ -95,6 +95,7 @@ int isEmpty(linked_list_t *list) {
 }
 
 void free_list(linked_list_t* list) {
+  if (list->head != 0 || list->head!=NULL) {return;}
```

Comment:
I don't think we should include this. then we don't know if the free was successful

Reply:
the assertion was giving errors before but that was before the free fix in maintenance so maybe assertion will be good now wasn’t before tho :/

Reply:
that means there's some issue with our memory management we should fix

Reply:
yea that was in maintenance should be good now i think

Reply:
tried to just an assert, no bueno :/ left the if statement

Reply:
Fixed the issue. The last thread running wasn't being freed from the list if it exited normally.